Assets/Scripts/CameraMovement.cs

```diff
@@ -0,0 +1,92 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraMovement : MonoBehaviour
+{
+    [SerializeField]
+    private float speed = 1f;
+
+    [SerializeField]
+    private Transform centrePoint;
+
+    private float _width;
+    private float _height;
+
+    private float _currentRadius;
+    private float _currentPolar;
+    private float _currentElevation;
+    private Vector3 _newPosition;
+
+    void Start()
+    {
+        _width = Screen.width / 2f;
+        _height = Screen.height / 2f;
+        CalculateStartingSphericalCoordinates();
+    }
+
+    void Update()
+    {
+        if (Input.touchCount > 0)
+        {
+            Touch touch = Input.GetTouch(0);
+            if (Input.touchCount == 2)
+            {
+                HandleZoom(touch);
+            }
+            else
+            {
+                MoveCamera(touch);
+            }
+            UpdateCartesian();
+            transform.position = _newPosition;
+            transform.LookAt(centrePoint, Vector3.up);
+        }
+    }
+
+    private void HandleZoom(Touch touch)
+    {
+        Touch touchOne = Input.GetTouch(1);
+        Vector2 touchZeroPreviousPosition = touch.position - touch.deltaPosition;
+        Vector2 touchOnePreviousPosition = touchOne.position - touchOne.deltaPosition;
+        float prevTouchDeltaMag = (touchZeroPreviousPosition - touchOnePreviousPosition).magnitude;
+        float TouchDeltaMag = (touch.position - touchOne.position).magnitude;
+        float deltaMagDiff = prevTouchDeltaMag - TouchDeltaMag;
+        _currentRadius += deltaMagDiff * Time.deltaTime;
```

Comment:
We could easily define a minimum and maximum radius here and prevent it from becoming unreasonable. The minimum value is especially important since we don't want the radius to be negative.

The implementation can be as simple as:
```suggestion
        _currentRadius = Mathf.Clamp(
            _currentRadius + deltaMagDiff * Time.deltaTime,
            minimumRadius,
            maximumRadius
        );
```
(`minimumRadius` and `maximumRadius` fields would need to be defined to apply this suggestion)